### PR TITLE
add `message_thread_id` param to methods, `message_thread_id` and `is_topic_message` to `Message`

### PR DESCRIPTION
**Pull request by lemontree210**
addresses #3340 

- [x] Add `message_thread_id` to methods in `Bot` and `ExtBot`
    - [x] add parameter
    - [x] add docstrings with `.. versionadded:: version` 
    - [x] add argument to a call to `send_message` or item to `data`
- [x] Add method to shortcuts (`Chat`, `User`...)
- [ ] Create new or adapted existing unit tests - **will only be possible when class for topics is created**

### CODE REVIEW

File: telegram/_bot.py [hunk 1]
```diff
@@ -965,6 +982,8 @@ async def send_photo(
                 receive a notification with no sound.
             protect_content (:obj:`bool`, optional): Protects the contents of the sent message from
                 forwarding and saving.
+            message_thread_id (:obj:`int`, optional): Unique identifier for the target message
+                thread (topic) of the forum; for forum supergroups only.
 
                 .. versionadded:: 13.10
```
Comment (Poolitzer) [inline telegram/_bot.py:988]: you sneaked the message_thread_id into the old versionadded

**Review by Poolitzer — CHANGES_REQUESTED** [review]
Looks good!

File: telegram/_bot.py (hunk 1 above)
Comment (lemontree210) [reply telegram/_bot.py:988]: D'oh. Thanks for spotting!

**Comment by lemontree210** [comment]
Hey guys, do you maybe think that we should merge this into `api_6.3_update` and then I'd branch off again when it becomes possible to add tests with real methods and attributes?

File: telegram/_message.py [hunk 2]
```diff
@@ -775,6 +795,7 @@ async def reply_text(
         allow_sending_without_reply: ODVInput[bool] = DEFAULT_NONE,
         entities: Union[List["MessageEntity"], Tuple["MessageEntity", ...]] = None,
         protect_content: ODVInput[bool] = DEFAULT_NONE,
+        message_thread_id: int = None,
```
Comment (Bibo-Joshi) [inline telegram/_message.py:798]: If the bot gets a message that `message` that has `message.message_thread_id`, what happens if we do 

```
send_message(
    chat_id=message.chat_id,
    text="hi",
    reply_to_message_id=message.message_id,
)
```

Does that make a reply in the correct topic? Or do we have to do 

```
send_message(
    chat_id=message.chat_id,
    text="hi",
    reply_to_message_id=message.message_id,
    message_thread_id=message.message_thread_id
)
```

? Can you try that out? How does the behavior differ between passing both `message_thread_id` and `reply_to_message_id` and only passing `message_thread_id`?

I'm trying to figure out, if passing `message_thread_id` should be automated through the `quote` argument 🤔  This doesn't have to be done in this PR though, that can be done on the api 6.3 branch also later on …

**Review by Bibo-Joshi — APPROVED** [review]
One inquery, but the changes LGTM :) can be merged IMO

File: telegram/_message.py (hunk 2 above)
Comment (Poolitzer) [reply telegram/_message.py:798]: > Does that make a reply in the correct topic?

yes

> if passing message_thread_id should be automated through the quote argument
not needed

**Pull request merged**